web/web.go

```diff
@@ -379,23 +389,15 @@ func New(logger log.Logger, o *Options) *Handler {
 			if p != rp {
 				continue
 			}
-
-			f, err := ui.Assets.Open("/static/react/index.html")
-			if err != nil {
-				w.WriteHeader(http.StatusInternalServerError)
-				fmt.Fprintf(w, "Error opening React index.html: %v", err)
-				return
-			}
-			idx, err := ioutil.ReadAll(f)
-			if err != nil {
-				w.WriteHeader(http.StatusInternalServerError)
-				fmt.Fprintf(w, "Error reading React index.html: %v", err)
-				return
+			if reactIdx == nil { // if for some reason the index didn't load, try again
+				loadReactIdx()
+				if reactIdx == nil { // if it's still not loaded, serve an error to the user
+					w.WriteHeader(http.StatusInternalServerError)
+					fmt.Fprintf(w, "Error reading React index.html: %v", err)
```

Comment:
AFAICS this `err` is not accessing the error from loading the React index, but some other randomly set error from above? I'd probably just keep the loading code where it is and thus keep loading the file on every request, as that is what's happening for all other assets as well. E.g. when you use the strategy of removing the `builtinassets` build tag to serve files from local disk (see https://github.com/prometheus/prometheus/blob/master/web/ui/README.md) during development (vs. using , you'd also want changes to this file to be picked up.